strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/layout/pypi/PypiArtifactController.java

```diff
@@ -0,0 +1,351 @@
+                            @ApiResponse(code = HttpURLConnection.HTTP_SEE_OTHER, message = "See Other Location"),
+                            @ApiResponse(code = HttpURLConnection.HTTP_NOT_FOUND, message = "Request Url Not Found"),
+                            @ApiResponse(code = HttpURLConnection.HTTP_INTERNAL_ERROR, message = "An error occurred while executing download request."),
+                            @ApiResponse(code = HttpURLConnection.HTTP_UNAVAILABLE, message = "Service Unavailable.") })
+    @PreAuthorize("hasAuthority('ARTIFACTS_RESOLVE')")
+    @RequestMapping(path = "{storageId}/{repositoryId}/{packageName}/", method = RequestMethod.GET)
+    public ResponseEntity<String> downloadPackage(@RepositoryMapping Repository repository,
+                                                  @PathVariable(name = "packageName") String packageName,
+                                                  HttpServletRequest request)
+    {
+
+        logger.info("install package request for storageId -> [{}] , repositoryId -> [{}], packageName -> [{}]",
+                    repository.getStorage().getId(),
+                    repository.getId(), packageName);
+
+        final Map<String, String> uriVariables = new HashMap<String, String>();
+        uriVariables.put("storageId", repository.getStorage().getId());
+        uriVariables.put("repositoryId", repository.getId());
+        uriVariables.put("packageName", packageName);
+
+        final URI location = ServletUriComponentsBuilder
+                                                        .fromCurrentServletMapping()
+                                                        .path("/storages/{storageId}/{repositoryId}/simple/{packageName}/")
+                                                        .build()
+                                                        .expand(uriVariables)
+                                                        .toUri();
+
+        final HttpHeaders headers = new HttpHeaders();
+        headers.setLocation(location);
+        return ResponseEntity.status(HttpStatus.SEE_OTHER)
+                             .headers(headers)
+                             .body(HttpStatus.SEE_OTHER.getReasonPhrase());
+    }
+
+    @ApiOperation(value = "This Endpoint will be used to retreive pypi artifact/.whl file")
+    @ApiResponses(value = { @ApiResponse(code = HttpURLConnection.HTTP_OK, message = "Success"),
+                            @ApiResponse(code = HttpURLConnection.HTTP_NOT_FOUND, message = "Request Url Not Found"),
+                            @ApiResponse(code = HttpURLConnection.HTTP_INTERNAL_ERROR, message = "An error occurred while executing download request."),
+                            @ApiResponse(code = HttpURLConnection.HTTP_UNAVAILABLE, message = "Service Unavailable.") })
+    @PreAuthorize("hasAuthority('ARTIFACTS_RESOLVE')")
+    @RequestMapping(path = "{storageId}/{repositoryId}/packages/{artifactName:.+}", method = RequestMethod.GET)
```

Comment:
@anki2189 why this `{artifactName:.+}` pattern needed here?

Reply:
Because name of file is passed in URL like 
`GET /storages/storage-pypi/pypi-releases/packages/pip_package_with_dependency-1.0-py3-none-any.whl HTTP/1.1\r\n`

Reply:
it should work without pattern with just `{artifactName}` for `/pip_package_with_dependency-1.0-py3-none-any.whl`